### ISSUE TITLE
RayShape and RayCast don't work as expected when returning contact normal (also Bullet and GodotPhysics inconsistencies))

### ISSUE THREAD

**Issue by n-pigeon** [godotengine/godot#12703]
**Godot version:** 386ec5d9549bb0ee922358ead86ebc7ad478b910

**Issue description**

**What happened (Bullet):**
**RayShape**: returns normal that is its own normal with different turn.
**RayCast**: same as RayShape.

-----------------

**What happened (GodotPhysics):**
**RayShape**: returns normal that is its own normal with different turn.
**RayCast**: Works as expected but only if RigidBody is set to **Character Mode**. If not in Character mode works the same as RayShape.

--------------------

**What was expected:**
**RayShape** when using "Contact Monitoring" returns **contact normal** of surface contact it monitored.
**RayCast** node returns **collision normal** returns normal of surface it collided with.
Both **RayCast** and **RayShape** return (with small error expected) matching result when they have the same position, direction and turn (not sure if turn is proper name in English, sorry :P).

Basically both RayShape and RayCast should work like - GodotPhysics RayCast when it is child of Character mode RigidBody.

---------------------------------------

**Link to minimal example project:**

[RayTests.zip](https://github.com/godotengine/godot/files/1446409/RayTests.zip)

**Comment by n-pigeon** [comment]
Ping: @AndreaCatania @reduz 
I've stumbled on this bug while testing my character controller.

I'm not sure how it was intended to work but it is inconsistent and doesn't work as expected, that is, not very useful.
When we try to retrieve contact point normal, we always want normal of the surface we are touching, not our ray swapped normal, which we can easily get because we know our Ray :D.

**Comment by AndreaCatania** [comment]
With bullet I can return the "surface" normal easily but is it the correct or it is the intended way how it should work? 

Are you sure that RayCast node return the Ray normal? this is odd

**Comment by n-pigeon** [comment]
@AndreaCatania 
I made test demo (bottom of the report) you can check it out.

I think rays should always return normal of surface they collide/contact with.
Ray normal would be really easy to get using math because we know our Ray.

We always use Rays to help us "see" the world :)

**Comment by AndreaCatania** [comment]
@n-pigeon After reduz talks we got that the ray shape should return always the normal of what it touches and not the normal from the hitten body.

If you think about it is more correct doing this, even if this could avoid another raycast check..

probably could be an idea create an API that allow to access to this information, but I'm not so sure if it worth.

**Comment by AndreaCatania** [comment]
Also I've tested your project in order to verify this:

> **What happened (Bullet):**
**RayShape**: returns normal that is its own normal with different turn.
**RayCast**: same as RayShape.

The raycast doesn't return the same normal of **RayShape**, it returns the normal of hit body as godot physics does. Can you please check it?

**Comment by akien-mga** [comment]
So if I understand correctly this is fixed/not a bug?

**Comment by AndreaCatania** [comment]
@akien-mga I don't know, I was not able to reproduce what @n-pigeon has reported. Probably it was fized with another fix.

@n-pigeon can you please check if you still have the problem?

**Comment by akien-mga** [comment]
I've tested the demo project and both Bullet and GodotPhysics return the same values, so I guess it's fixed.
